sql/moz-fx-data-shared-prod/bigeye_derived/dashboard_service_v1/query.py

```diff
@@ -0,0 +1,149 @@
+    "DASHBOARD_DATA_POINT_TYPE_PCT_ISSUES_INTERACTED",
+    "DASHBOARD_DATA_POINT_TYPE_PCT_TABLES_WITHOUT_ISSUE_BY_CATEGORY",
+    "DASHBOARD_DATA_POINT_TYPE_TOTAL_TABLES_WITHOUT_ISSUE_BY_CATEGORY",
+]
+
+
+def make_api_request(workspace_id: int) -> Dict[str, Any]:
+    """Make API request to Bigeye for a specific workspace."""
+    payload = {
+        "aggregationType": "DASHBOARD_DATE_AGGREGATION_TYPE_DAY",
+        "dataPointTypes": DATA_POINT_TYPES,
+        "workspaceIds": [workspace_id],
+    }
+
+    headers = {
+        "accept": "application/json",
+        "content-type": "application/json",
+        "authorization": BIGEYE_API_KEY,
+    }
+
+    response = requests.post(API_URL, json=payload, headers=headers, timeout=30)
+    response.raise_for_status()
+    return response.json()
+
+
+def process_response(response_data: Dict[str, Any]) -> pd.DataFrame:
+    """Process API response into a DataFrame."""
+    df = pd.json_normalize(
+        response_data["dataPointSeries"],
+        record_path="dataPoints",
+        meta=["dataPointType"],
+        errors="ignore",
+    )
+    df["workspaceIds"] = df["workspaceIds"].str[0]
+
+    df.drop(columns=["sourceIds"], errors="ignore", inplace=True)
+    df["refreshed_at"] = pd.Timestamp.now(tz="UTC")
+
+    df = df.rename(
+        columns={
+            "dataPointType": "data_point_type",
+            "workspaceIds": "workspace_id",
+            "dataTimestamp": "timestamp",
+        }
+    )
+
+    df["timestamp"] = pd.to_datetime(df["timestamp"], unit="s")
+
+    return df
+
+
+def get_bigeye_data() -> pd.DataFrame:
+    """Fetch data from Bigeye API for all workspaces and combine into a single DataFrame."""
+    all_data = []
+
+    for workspace_id in WORKSPACE_IDS:
+        try:
+            response_data = make_api_request(workspace_id)
+            df = process_response(response_data)
+            all_data.append(df)
+            logging.info(f"Successfully processed data for workspace {workspace_id}")
+        except Exception as e:
+            logging.info(f"Error processing workspace {workspace_id}: {str(e)}")
+            continue
+
+    if not all_data:
+        raise Exception("No data was successfully processed from any workspace")
+
+    return pd.concat(all_data, ignore_index=True)
+
+
+def validate_dataframe(df: pd.DataFrame) -> None:
+    """Perform basic validation on the DataFrame."""
+    required_columns = {
+        "data_point_type",
+        "timestamp",
+        "workspace_id",
+        "value",
+        "refreshed_at",
+    }
+    if not required_columns.issubset(df.columns):
+        missing = required_columns - set(df.columns)
+        raise ValueError(f"DataFrame is missing required columns: {missing}")
+
+
+def load_to_bigquery(project_id, df: pd.DataFrame) -> None:
+    """Load DataFrame to BigQuery."""
+    client = bigquery.Client(project_id)
+
+    job = client.load_table_from_dataframe(
+        df,
+        TARGET_TABLE,
```

Comment:
I'd suggest we make targaet_table a param in this function and pass it in as an argument to make it more flexible in case we want to write to a different target table.